src/components/docs-version-switcher/index.tsx

```diff
@@ -45,71 +39,63 @@ const DocsVersionSwitcher = ({
 		return null
 	}
 
-	const latestVersion = options.find(
-		(option: VersionSelectItem) => option.isLatest === true
-	)
-
-	// Get the selected option
+	/**
+	 * Find the selected option, either by parsing a version from the path,
+	 * or fallback to latest (for non-versioned URLs).
+	 */
 	const selectedVersion = getVersionFromPath(currentPath)
 	let selectedOption: DocsVersionSwitcherOption
 	if (selectedVersion) {
 		selectedOption = options.find(
 			(option: DocsVersionSwitcherOption) => option.version === selectedVersion
 		)
 	} else {
-		selectedOption = latestVersion
+		// Fall back to selecting the latest version
+		selectedOption = options.find(
+			(option: VersionSelectItem) => option.isLatest === true
+		)
+	}
+
+	/**
+	 * Construct a project name to be used in ariaLabels for each option.
+	 */
+	let projectNameForLabel: string
+	if (projectName) {
+		projectNameForLabel = projectName
+	} else {
+		const docsName = currentRootDocsPath.shortName || currentRootDocsPath.name
+		projectNameForLabel = `${currentProduct.name} ${docsName}`
 	}
 
-	// Build the `ariaLabel` that is announced when the activator is focused
-	const currentRootDocsPathName =
-		currentRootDocsPath.shortName || currentRootDocsPath.name
-	const nameForLabel =
-		projectName || `${currentProduct.name} ${currentRootDocsPathName}`
-	const ariaLabel = `Choose a ${nameForLabel} version. Currently viewing ${selectedOption.label}.`
+	/**
+	 * Encode docs concerns into the `options` to pass to `VersionSwitcher`.
+	 */
+	const optionsForVersionSwitcher = options.map(
```

Comment:
`DocsVersionSwitcher` has been updated to render `VersionSwitcher` under the hood. This logic was pulled out of the previous `DocsVersionSwitcher` render body.